### PR TITLE
Rewrite Sink.ignore as a GraphStage

### PR DESCRIPTION
**Pull request by ortibazar**

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

-- 
Konrad Malawski

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 1]
```diff
@@ -329,6 +329,42 @@ object GraphStages {
   }
 
   /**
+   * INTERNAL API
+   * Discards all received elements.
+   */
+  lazy val IgnoreSink = new GraphStageWithMaterializedValue[SinkShape[Any], Future[Done]] {
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Not sure it is worth it to have it lazy

**Review by johanandren — COMMENTED** [review]
Is there any pre-existing test coverage (even though it is a very simple stage)?

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala (hunk 1 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: It doesn't worth it. Ignore is used a lot.

**Review by drewhk — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
LGTM after https://github.com/akka/akka/pull/21728/files#r85068270 addressed

**Comment by ktoso** [comment]
Great, thanks :-)

**Review by ktoso — APPROVED** [review]

**Comment by johanandren** [comment]
Refs #21527

**Comment by johanandren** [comment]
Thanks @ortigali

**Pull request merged**